From 32e4be60b8fa8799927f6e0a6f698db0d781b369 Mon Sep 17 00:00:00 2001
From: Goh Wei Wen <goweiwen@gmail.com>
Date: Wed, 19 Jul 2023 19:47:35 +0800
Subject: [PATCH] feat: retain fast-forward state when pausing

---
 runloop.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/runloop.c b/runloop.c
index 55313076f0..d264d5989f 100644
--- a/runloop.c
+++ b/runloop.c
@@ -6476,13 +6476,13 @@ static enum runloop_state_enum runloop_check_state(
       if (!check2)
          check2 = old_hold_button_state != new_hold_button_state;

-      /* Don't allow fastmotion while paused */
-      if (check2 && runloop_paused)
-      {
-         new_button_state      = false;
-         new_hold_button_state = false;
-         input_st->flags      |= INP_FLAG_NONBLOCKING;
-      }
+      // /* Don't allow fastmotion while paused */
+      // if (check2 && runloop_paused)
+      // {
+      //    new_button_state      = false;
+      //    new_hold_button_state = false;
+      //    input_st->flags      |= INP_FLAG_NONBLOCKING;
+      // }

 #ifdef HAVE_NETWORKING
       if (check2 && !netplay_allow_timeskip)
