diff --git a/tasks/task_save.c b/tasks/task_save.c
index 0f867f38d0..ae6d252fa6 100644
--- a/tasks/task_save.c
+++ b/tasks/task_save.c
@@ -793,11 +793,7 @@ static void task_load_handler(retro_task_t *task)
       {
          char msg[128];

-         if (state->flags & SAVE_TASK_FLAG_AUTOLOAD)
-            snprintf(msg, sizeof(msg),
-                  msg_hash_to_str(MSG_AUTOLOADING_SAVESTATE_SUCCEEDED),
-                  path_basename(state->path));
-         else
+         if (!(state->flags & SAVE_TASK_FLAG_AUTOLOAD))
          {
             if (state->state_slot < 0)
                strlcpy(msg,
@@ -807,9 +803,8 @@ static void task_load_handler(retro_task_t *task)
                snprintf(msg, sizeof(msg),
                      msg_hash_to_str(MSG_LOADED_STATE_FROM_SLOT),
                      state->state_slot);
+            task_set_title(task, strdup(msg));
          }
-
-         task_set_title(task, strdup(msg));
       }

       goto end;
